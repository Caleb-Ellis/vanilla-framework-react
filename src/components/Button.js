import React from 'react';

class Button extends React.Component {
  render() {
    return (
      <p><button className={"p-button--" + this.props.type}>{this.props.value}</button></p>
    )
  }
}

export default Button;
