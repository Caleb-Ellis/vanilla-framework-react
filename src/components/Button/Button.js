import React from 'react';
import './Button.css';

class Button extends React.Component {
  render() {
    return (
      <button className={`p-button--${this.props.type}`}>{this.props.value}</button>
    )
  }
}

Button.defaultProps = {type: 'base'};

export default Button;
