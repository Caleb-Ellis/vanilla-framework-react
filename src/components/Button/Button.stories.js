import React from 'react';
import { storiesOf } from '@storybook/react';

import Button from './Button';

storiesOf('Buttons', module)
  .add('Base button', () => (
    <Button value="Base button 😶" />
  ))
  .add('Neutral button', () => (
    <Button value="Neutral button 😐" modifier="neutral" />
  ))
  .add('Brand button', () => (
    <Button value="Brand button 💥" modifier="brand" />
  ))
  .add('Negative button', () => (
    <Button value="Negative button 😡" modifier="negative" />
  ))
  .add('Positive button', () => (
    <Button value="Positive button 😁" modifier="positive" />
  ));
