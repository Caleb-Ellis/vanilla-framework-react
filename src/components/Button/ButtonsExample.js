import React from 'react';
import Button from './Button';
import './Button.css';

class ButtonsExample extends React.Component {
  render() {
    return (
      <div>
        <h2>Buttons</h2>
        <p><Button value="Base button 😶" /></p>
        <p><Button value="Neutral button 😐" modifier="neutral" /></p>
        <p><Button value="Brand button 💥" modifier="brand" /></p>
        <p><Button value="Negative button 😡" modifier="negative" /></p>
        <p><Button value="Positive button 😁" modifier="positive" /></p>
      </div>
    );
  }
}

Button.defaultProps = { modifier: 'base' };

export default ButtonsExample;
