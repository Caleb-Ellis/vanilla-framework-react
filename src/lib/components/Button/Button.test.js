import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Button from './Button';

describe('Button component', () => {
  it('should compare with a snapshot', () => {
    const button = ReactTestRenderer.create(<Button value="Submit" />);
    const json = button.toJSON();
    expect(json).toMatchSnapshot();
  });
});

