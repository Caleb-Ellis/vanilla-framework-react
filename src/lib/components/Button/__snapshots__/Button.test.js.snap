// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should compare with a snapshot 1`] = `
<button
  className="p-button--base"
>
  Submit
</button>
`;
